xgi/generators/uniform.py

```diff
@@ -286,10 +341,26 @@ def uniform_erdos_renyi_hypergraph(n, m, p, p_type="degree", seed=None):
     Hypergraph
         The Erdos Renyi hypergraph
 
-
     See Also
     --------
     ~xgi.generators.random.random_hypergraph
+
+    Notes
+    -----
+    Because XGI only stores edges as sets, if self-loops are allowed,
+    for example, the edge (0, 0, 0) will be mapped to {0}. However,
+    because this is explicitly a *uniform* method, we discard these edges
+    so that this is the case. For sparse networks, this is a rare
+    occurrence and this method offers an order of magnitude speedup,
+    so while it is not the default behavior, this option is exposed to
```

Comment:
The option of doing what? Sorry I don't understand how the multiedge option is related to what's above in this Note 🥲 I'll try so maybe you'll understand what I don't understand hehe:

For sparse networks, we need to discard fewer edges that are self-loops (result in in edges of size < m), simply because we produce few edges overall. Because of that, this method (meaning, the way of generating edges with the geometric series, but only with multiedge=True?) is faster (compared to the naive other random hypergraph function or compared to multiedge=False?). So, the we give the option to set multiedge=True, even though the default is multiedge=False. 

Are we saying multiedge=True, is faster, and in sparse networks we will need to discard fewer self-loops (in absolute numbers, not relatively to the total number of edges though) so that it makes proportionally even faster in that case?

Reply:
Hopefully my revision addressed this! Basically what I'm saying is that (1) allowing self-loops and multiedges is much faster and (2) when the hypergraph is sparse, not many artifacts are created anyway so it's not so bad.